workflows/snvphylnfc.nf

```diff
@@ -67,34 +98,118 @@ workflow SNVPHYL {
     )
-    ch_versions = ch_versions.mix(ASSEMBLY_STUB.out.versions)
+    ch_versions = ch_versions.mix(SMALT_MAP.out.versions)
 
-    // A channel of tuples of ({meta}, [read[0], read[1]], assembly)
-    ch_tuple_read_assembly = input.join(ASSEMBLY_STUB.out.assembly)
+    SORT_INDEX_BAMS(
+        SMALT_MAP.out.bams
+    )
+    ch_versions = ch_versions.mix(SORT_INDEX_BAMS.out.versions)
+
+    // TODO: Review this module further (purpose, version, code injection, container, etc.)
+    GENERATE_LINE_1(
+        SORT_INDEX_BAMS.out.sorted_bams.collect()
+    )
+
+    VERIFYING_MAP_Q(
+        SORT_INDEX_BAMS.out.sorted_bams.collect(), GENERATE_LINE_1.out.bam_lines_file.splitText()
+    )
+    ch_versions = ch_versions.mix(VERIFYING_MAP_Q.out.versions)
+
+    FREEBAYES(
+        SORT_INDEX_BAMS.out.sorted_bams_and_sampleID, params.refgenome
+    )
+    ch_versions = ch_versions.mix(FREEBAYES.out.versions)
+
+    FILTER_FREEBAYES(
+        FREEBAYES.out.vcf_files
+    )
+    ch_versions = ch_versions.mix(FILTER_FREEBAYES.out.versions)
 
-    GENERATE_SAMPLE_JSON (
-        ch_tuple_read_assembly
+    // TODO: Is there a better way of doing this? In the last step?
+    BGZIP_FREEBAYES_VCF(
+        FILTER_FREEBAYES.out.filtered_vcf
     )
-    ch_versions = ch_versions.mix(GENERATE_SAMPLE_JSON.out.versions)
+    ch_versions = ch_versions.mix(BGZIP_FREEBAYES_VCF.out.versions)
 
-    GENERATE_SUMMARY (
-        ch_tuple_read_assembly.collect{ [it] }
+    FREEBAYES_VCF_TO_BCF(
+        BGZIP_FREEBAYES_VCF.out.filtered_zipped_vcf
     )
-    ch_versions = ch_versions.mix(GENERATE_SUMMARY.out.versions)
+    ch_versions = ch_versions.mix(FREEBAYES_VCF_TO_BCF.out.versions)
 
-    SIMPLIFY_IRIDA_JSON (
-        GENERATE_SAMPLE_JSON.out.json
+    MPILEUP(
+        SORT_INDEX_BAMS.out.sorted_bams_and_sampleID, params.refgenome
     )
-    ch_versions = ch_versions.mix(SIMPLIFY_IRIDA_JSON.out.versions)
-    ch_simplified_jsons = SIMPLIFY_IRIDA_JSON.out.simple_json.map { meta, data -> data }.collect() // Collect JSONs
+    ch_versions = ch_versions.mix(MPILEUP.out.versions)
 
-    IRIDA_NEXT_OUTPUT (
-        samples_data=ch_simplified_jsons
+    BGZIP_MPILEUP_VCF(
+        MPILEUP.out.mpileup
     )
-    ch_versions = ch_versions.mix(IRIDA_NEXT_OUTPUT.out.versions)
+    ch_versions = ch_versions.mix(BGZIP_MPILEUP_VCF.out.versions)
+
+    BCFTOOLS_CALL(
+        BGZIP_MPILEUP_VCF.out.mpileup_zipped
+    )
+    ch_versions = ch_versions.mix(BCFTOOLS_CALL.out.versions)
+
+    //Joining channels of multiple outputs
+    combined_ch = BCFTOOLS_CALL.out.mpileup_bcf.join(FREEBAYES_VCF_TO_BCF.out.filtered_bcf)
+    //11. consolidate variant calling files process takes 2 input channels as arguments
+    CONSOLIDATE_BCFS(
+        combined_ch
+    )
+    ch_versions = ch_versions.mix(CONSOLIDATE_BCFS.out.versions)
+
+    // Concat filtered densities to make new invalid_postions
+    CONSOLIDATE_FILTERED_DENSITY(
```

Comment:
There's an nf-core module `cat_cat` which could likely replace this: https://nf-co.re/modules/cat_cat

Reply:
Updated in [4ed4f8b](https://github.com/phac-nml/snvphylnfc/pull/3/commits/4ed4f8bd1250f2973f26b630787a345d8af76f34)